app/main.py

```diff
@@ -76,3 +102,131 @@ def get_share_link():
 @app.route("/state/getstate", methods=["POST"])
 def get_state():
     return get_saved_state(statetable)
+
+
+@app.route("/spreadsheet")
+# Connect to the google spreadsheet and get all spreadsheet data.
+def spreadsheet():
+    scope = [
+        "https://spreadsheets.google.com/feeds", "https://www.googleapis.com/auth/spreadsheets",
+        "https://www.googleapis.com/auth/drive.file", "https://www.googleapis.com/auth/drive"
+    ]
+    credential = ServiceAccountCredentials.from_json_keyfile_dict(
+        SPREADSHEET_CREDENTIALS, scope)
+    client = gspread.authorize(credential)
+    gsheet = client.open("organ_sheets").sheet1
+    data = gsheet.get_all_records()
+    return jsonify(data)
+
+
+@app.route("/program", methods=['GET', 'POST'])
+# Get the program information from Gen3 Data Commons
+def program():
+    res = requests.get(
+        f'{Gen3Config.GEN3_ENDPOINT_URL}/api/v0/submission/', headers=HEADER)
+
+    json_data = json.loads(res.content)
+    program_list = []
+    for ele in json_data['links']:
+        program_list.append(ele.replace(
+            "/v0/submission/", ""))
+    new_json_data = {'program': program_list}
+    return new_json_data
+
+
+@app.route("/<program>/project", methods=['GET'])
+# Get all projects information from Gen3 Data Commons
+def project(program):
+    res = requests.get(
+        f'{Gen3Config.GEN3_ENDPOINT_URL}/api/v0/submission/{program}', headers=HEADER)
+
+    json_data = json.loads(res.content)
+    project_list = []
+    for ele in json_data['links']:
+        project_list.append(ele.replace(
+            f"/v0/submission/{program}/", ""))
+    new_json_data = {'project': project_list}
+    return new_json_data
+
+
+@app.route("/dictionary", methods=['GET', 'POST'])
+# Get all dictionary node from Gen3 Data Commons
+def dictionary():
+    res = requests.get(
+        f'{Gen3Config.GEN3_ENDPOINT_URL}/api/v0/submission/_dictionary', headers=HEADER)
+
```

Comment:
Should handle the case when the requests.get fails, in which case the res.content will not be valid and will cause uncaught exception in the backend.